sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -59,21 +65,16 @@
   // The 4-bytes header of `numElements` is also included.
   private int sizeInBytes;
 
-  public Object getBaseObject() { return baseObject; }
-  public long getBaseOffset() { return baseOffset; }
-  public int getSizeInBytes() { return sizeInBytes; }
+  /** The width of the null tracking bit set plus `numElements`, in bytes */
+  private int headerInBytes;
```

Comment:
It's more often to use `baseOffset + headerInBytes`, we could have elementOffset instead of headerInBytes.